agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2225,10 +2225,11 @@ bool backhaul_manager::handle_1905_autoconfiguration_response(ieee1905_1::CmduMe
                     LOG(DEBUG) << FSM_CURR_STATE_STR;
                     soc->controller_discovered = true;
                     controller_bridge_mac      = src_mac;
-                    // the backhaul was operational on slave registration.
+                    // the backhaul was operational on slave registration or changed
+                    // to operational since the registration.
                     // This means that it was restarted and sending backhaul connected event
                     // is required so that the slave will re-join the controller.
-                    if (soc->operational_on_registration) {
+                    if (soc->operational_on_registration || FSM_IS_IN_STATE(OPERATIONAL)) {
```

Comment:
This change reverts commit https://github.com/prplfoundation/prplMesh/commit/17ad041b5eb40b790d08d0e8ba225fe829d134e0.

The reason we shouldn't do it is that once we got the first discovery response message we would move to an operational state and send the "backhaul connected" message to all son_slave which are discovered.
This means that after we will get the second discovery response we would get inside the "finalize" function and send this message again. This means that the previous son_slave which we got the discovery response for, will receive a "backhaul connected" message twice.

Reply:
Ok, I didn't notice I was actually reverting a previous change.
Thanks for the explanation.
And what if we only sent the "backhaul connected" message only when we have the autoconfig response and not when moving to the operational state? Does it make any sense? 
I'll push the change to this branch so that you can see what I mean.

Reply:
>This means that the previous son_slave which we got the discovery response for, will receive a "backhaul connected" message twice.

So? The slave can simply ignore the second message. This is merely an optimization, and this PR proves it was wrong since it didn't cover all cases.

I think that instead of letting Raphael struggle with a new solution we should revert the optimization and be done with it.

Reply:
This 

> > This means that the previous son_slave which we got the discovery response for, will receive a "backhaul connected" message twice.
> 
> So? The slave can simply ignore the second message. This is merely an optimization, and this PR proves it was wrong since it didn't cover all cases.
> 
> I think that instead of letting Raphael struggle with a new solution we should revert the optimization and be done with it.
> 
> @arnout @vitalybu what do you think?

This is actually bad because when the son_slave will receive this message it will change its state to a different state and send messages it already sent.
Don't forget, we both did this because we noticed it causing a mess on the son_slave and the logs too.
BTW, this is one of the reasons we need unified Agent and not two independent "son_slaves".

Reply:
So we should add a check for that in the son slave - if it is already operational it can ignore the second message.
I know that this is probably not the best solution so if you have another one please speak up, we need this resolved *today*

Reply:
@tomereli @morantr could you comment on commit ce621129ae9adf2ff0acc822bd544e0adf291206 ? 
Why not call the finalize function when receiving the autoconfig reply only?
It avoids the duplicate M1 and shouldn't put the slave in a wrong state as far as I understand.

Reply:
I don't remember why we did it like this, but your suggestion might be a good idea.
I think that the logic behinds it is not to hold back auto-config flow if the other radio hasn't been answered, but basically I don't see any reason why would it happen.